beater/api/mux.go

```diff
@@ -181,18 +184,35 @@ func (r *routeBuilder) rumIntakeHandler(newProcessor func(*config.Config) *strea
 		requestMetadataFunc = rumRequestMetadata
 	}
 	return func() (request.Handler, error) {
-		batchProcessor := r.batchProcessor
+		var batchProcessors modelprocessor.Chained
+		if len(r.cfg.RumConfig.AllowServiceNames) > 0 {
+			batchProcessors = append(batchProcessors, allowedServiceNamesBatchProcessor(r.cfg.RumConfig.AllowServiceNames))
+		}
 		if r.sourcemapStore != nil {
-			batchProcessor = modelprocessor.Chained{
-				sourcemap.BatchProcessor{
-					Store:   r.sourcemapStore,
-					Timeout: r.cfg.RumConfig.SourceMapping.Timeout,
-				},
-				batchProcessor,
+			batchProcessors = append(batchProcessors, sourcemap.BatchProcessor{
+				Store:   r.sourcemapStore,
+				Timeout: r.cfg.RumConfig.SourceMapping.Timeout,
+			})
+		}
+		if r.cfg.RumConfig.LibraryPattern != "" {
+			re, err := regexp.Compile(r.cfg.RumConfig.LibraryPattern)
+			if err != nil {
+				return nil, errors.Wrap(err, "invalid library pattern regex")
+			}
+			batchProcessors = append(batchProcessors, modelprocessor.SetLibraryFrame{Pattern: re})
+		}
+		if r.cfg.RumConfig.ExcludeFromGrouping != "" {
+			re, err := regexp.Compile(r.cfg.RumConfig.ExcludeFromGrouping)
+			if err != nil {
+				return nil, errors.Wrap(err, "invalid exclude from grouping regex")
 			}
+			batchProcessors = append(batchProcessors, modelprocessor.SetExcludeFromGrouping{Pattern: re})
 		}
-		batchProcessor = batchProcessorWithAllowedServiceNames(batchProcessor, r.cfg.RumConfig.AllowServiceNames)
-		h := intake.Handler(newProcessor(r.cfg), requestMetadataFunc, batchProcessor)
+		if r.sourcemapStore != nil {
+			batchProcessors = append(batchProcessors, modelprocessor.SetCulprit{})
+		}
```

Comment:
Could you add a note that the order of the processors is important?

Reply:
done